refactored: classes and endpoints to ensure consistency with the database design

kostiantynKarzhanov opened the pull request.

Commit:
refactored: 'users' -> 'user' endpoint and DB table

Commit:
refactored: 'attendanceRecord' -> 'attendance' to be the same as endpoint

konstantinKarzhanov commented:
refactored: 

- 'users' -> 'user' endpoint and DB table
- 'attendanceRecord' -> 'attendance' to be the same as endpoint

Commit:
refactored: 'user_roles'->'user_role','UserRoles'->'UserRole'

Commit:
refactored: '/courses' -> '/course'

Commit:
refactored: '/instructors' -> '/instructor'

Commit:
refactored: '/students' -> '/student'

Commit:
modified: updated endpoints

BLewis-Keyin commented:
Going to hold off on this PR until after the presentation today.

BLewis-Keyin reviewed (approved):
The conflicts appear to just be some sql files I removed and replaced with a single startup sql script. This looks good though.

Commit:
resolved merge conflict

new: files in 'sql/dump' and 'sql/startscript.sql'

removed: old files in '/sql/'

removed: repeated 'spring-boot-starter-test' dependency

removed: unused imports in 'Instructor' entity
